### PR TITLE
Participant qualification

### PR DESCRIPTION
**Pull request by cytadela8**
Adds info about qualification and qualification buttons to profile page. 
+ Refactored permissions

### CODE REVIEW

File: wwwapp/views.py [hunk 1]
```diff
@@ -465,9 +481,8 @@ def unregister_from_workshop_view(request):
     return JsonResponse(data)
 
 
+@user_passes_test(is_superuser)
```
Comment (krzys-h) [inline wwwapp/views.py]: Remove this function in favor of a proper permission

File: wwwapp/views.py [hunk 2]
```diff
@@ -261,29 +278,29 @@ def can_edit_workshop(workshop, user):
...
 
-    if not can_see_all and not can_see_this:
-        # it should show page like "you don't have permission", probably
-        return redirect('login')
+    if not can_see:
+        return HttpResponseForbidden()
 
     context = get_context(request)
 
-    context['workshop'] = workshop
```
Comment (krzys-h) [inline wwwapp/views.py]: `_workshopeditheader.html` requires the workshop object

File: wwwapp/models.py [hunk 3]
```diff
@@ -37,16 +37,23 @@ def __str__(self):
         return "{0.first_name} {0.last_name}".format(self.user)
 
     class Meta:
-        permissions = (('see_all_users', 'Can see all users'),)
+        permissions = [('see_all_users', 'Can see all users'),
+                       ('export_workshop_registration', 'Can download workshop registration data')]
```
Comment (krzys-h) [inline wwwapp/models.py]: Migration probably required here

File: templates/profile.html [hunk 4]
```diff
@@ -72,10 +71,49 @@ <h1>{{ title }}</h1>
...
+                    <span class="not-qualified">✘{{ edition.year }}</span>
+                  {% endif %}
+                {% endfor %}
+                {% if has_workshop_profile %}
+                  <form method="post" class="inline">
+                    {% csrf_token %}
+                    <span class="btn-group-xs" role="group">
+                      <button type="submit" class="btn btn-default"
+                              name="qualify" value="delete"
+                              onclick="return confirm('Czy na pewno chcesz usunąć kwalifikację?');">
```
Comment (grzegorz225) [inline templates/profile.html]: https://stackoverflow.com/questions/8982295/confirm-deletion-in-modal-dialog-using-twitter-bootstrap

File: wwwapp/models.py [hunk 5]
```diff
@@ -244,6 +251,9 @@ def qualified_count(self):
             return None
         return self.workshopparticipant_set.filter(qualification_result__gte=self.qualification_threshold).count()
 
+    def is_accepted(self):
+        return self.status == 'Z'
```
Comment (grzegorz225) [inline wwwapp/models.py]: Change to STATUS_ACCEPTED

File: wwwapp/views.py [hunk 6]
```diff
@@ -592,12 +598,9 @@ def render_workshops(request, title, workshops):
     return render(request, 'workshoplist.html', context)
 
 
+@login_required()
+@permission_required('wwwapp.see_all_workshops', raise_exception=True)  # think about seperate permission
```
Comment (grzegorz225) [inline wwwapp/views.py]: Start sentence in comment from capital letter.

File: wwwapp/views.py [hunk 7]
```diff
@@ -593,12 +599,9 @@ def render_workshops(request, title, workshops):
...
+@permission_required('wwwapp.see_all_workshops', raise_exception=True)  # Think about seperate permission
 def emails_view(request):
-    # think about seperate permission
-    if not request.user.has_perm('wwwapp.see_all_workshops'):
-        # it should show page like "you don't have permission", probably
-        return redirect('login')
-
     workshops = Workshop.objects.all()
 
     result = []
```
Comment (radrow) [inline wwwapp/views.py]: `result = []`  – what's that?

Comment (cytadela8) [reply wwwapp/views.py]: This is returned.

**Pull request merged**